src/main/java/com/gotocompany/firehose/config/converter/GrpcMetadataConverter.java

```diff
@@ -0,0 +1,23 @@
+package com.gotocompany.firehose.config.converter;
+
+import org.aeonbits.owner.Converter;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import io.grpc.Metadata;
+
+public class GrpcMetadataConverter implements Converter<Metadata> {
+
+    @Override
+    public Metadata convert(Method method, String input) {
+
+        Metadata metadata = new Metadata();
+
+        Arrays.stream(input.split(",")).filter(metadataKeyValue -> !metadataKeyValue.trim().isEmpty()).forEach(metadataKeyValue -> {
+            if (metadataKeyValue.contains(":")) {
+                metadata.put(Metadata.Key.of(metadataKeyValue.split(":")[0], Metadata.ASCII_STRING_MARSHALLER), metadataKeyValue.split(":")[1]);
```

Comment:
if you apply the above split initially, you can avoid multiple splits in this calls.

Reply:
Also, is it good to strip the the beginning & trailing spaces of key and value? whats the convention followed across firehose, I'm not sure. Take a call on that.

Reply:
made changes